webapp/src/pages/app/projects/[projectName]/repos/[repoName]/list/[revision].tsx

```diff
@@ -47,9 +50,37 @@ const RepositoryDetailPage = () => {
   );
   const { isOpen, onOpen, onClose } = useDisclosure();
   const [tabIndex, setTabIndex] = useState(0);
+  const dispatch = useAppDispatch();
+
   const handleTabChange = (index: number) => {
     setTabIndex(index);
   };
+
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      await dispatch(createMessage({ title: '', text: 'copied to clipboard', type: 'success' }));
+    } catch (err) {
+      const error: string = ErrorHandler.handle(err);
+      await dispatch(createMessage({ title: 'failed to copy to clipboard', text: error, type: 'error' }));
+    } finally {
+      dispatch(resetState());
+    }
+  };
+
+  const constructApiUrl = (project: string, repo: string, path: string): string => {
+    return `${window.location.origin}/api/v1/projects/${project}/repos/${repo}/contents${path}`;
+  };
+  const handleCopyApiUrl = async (project: string, repo: string, path: string) => {
+    const apiUrl: string = constructApiUrl(project, repo, path);
+    copyToClipboard(apiUrl);
+  };
+  const handleCopyAsCurlCommand = async (project: string, repo: string, path: string) => {
+    const apiUrl: string = constructApiUrl(project, repo, path);
+    const curlCommand = `curl -XGET "${apiUrl}"`;
```

Comment:
We need to set `sessionId` as an access token to Authorization header to access resources using REST API.
```
curl -H "Authorization: Bearer access-token"
```

Reply:
on it

Reply:
added @ikhoon

Reply:
This is the copy result:

```shell
curl -XGET "http://localhost:3000/api/v1/projects/abcd/repos/repo1/contents/0-dwarf-hotot" \
-H "Authorization: Bearer <access-token>"
```